bluemira/equilibria/flux_surfaces.py

```diff
@@ -872,3 +876,251 @@ def poloidal_angle(Bp_strike, Bt_strike, gamma):
+    """
+    Returns spherical harmonics coefficients/amplitudes (A_l) to be used
+    in a spherical harmonic approximation of the vacuum/coil contribution
+    to the poilodal flux (psi). Vacuum Psi = Total Psi - Plasma Psi.
+    These coefficients can be used as contraints in optimisation.
+
+    For a single filiment (coil):
+
+        A_l =  1/2 * mu_0 * I_f * sin(theta_f) * (r_t/r_f)**l *
+                    ( P_l * cos(theta_f) / sqrt(l*(l+1)) )
+
+    Where l = degree, and P_l * cos(theta_f) are the associated
+    Legendre polynomials of degree l and order (m) = 1.
+
+    Parmeters
+    ----------
+    x_f: np.array
+        X coordinates of filaments (coils)
+    z_f: np.array
+        Z coordinates of filaments (coils)
+    i_f: np.array
+        Currents of filaments (coils)
+    max_degree: integer
+        Maximum degree of harmonic to calculate up to
+    r_t: float
+        Typical length scale (e.g. radius at outer midplane)
+
+    Returns
+    -------
+    amplitudes: np.array
+        Array of spherical harmonic amplitudes from given coil potitions and currents
+     max_valid_r: float
+        Maximum spherical radius for which the spherical harmonics apply
+    """
+    # SH coeffcients from fuction of the current distribution outside of the sphere
+    # containing the plamsa, i.e., LCFS (r_lcfs)
+    # SH coeffs = currents2harmonics @ coil currents
+    # N.B., max_valid_r >= r_lcfs,
+    # i.e., cannot use coil located within r_lcfs as part of this method.
+    currents2harmonics, max_valid_r = coil_harmonic_amplitude_matrix(
+        x_f, z_f, max_degree, r_t
+    )
+
+    return currents2harmonics @ i_f, max_valid_r
+
+
+def coil_harmonic_amplitude_matrix(x_f, z_f, max_degree, r_t):
+    """
+    Construct matrix from harmonic amplitudes at given coil locations.
+
+    Parmeters
+    ----------
+    x_f: np.array
+        X coordinates of filaments (coils)
+    z_f: np.array
+        Z coordinates of filaments (coils)
+    max_degree: integer
+        Maximum degree of harmonic to calculate up to
+    r_t: float
+        Typical length scale (e.g. radius at outer midplane)
+
+    Returns
+    -------
+    currents2harmonics: np.array
+        Matrix of harmonic amplitudes (to get spherical harmonic coefficents
+        -> matrix @ vector of coil currents, see coil_harmonic_amplitudes)
+     max_valid_r: float
+        Maximum spherical radius for which the spherical harmonic
+        approximation is valid
+    """
+    # Spherical coords
+    r_f = np.sqrt(x_f**2 + z_f**2)
+    theta_f = np.arctan2(x_f, z_f)
+    # Maxmimum r value for the sphere whithin which harmonics apply
+    max_valid_r = np.amin(r_f)
+
+    # [number of degrees, number of coils]
+    currents2harmonics = np.zeros([max_degree, np.size(r_f)])
+    # First 'harmonic' is constant (this line avoids Nan isuues)
+    currents2harmonics[0, :] = 1  #
+
+    # SH coeffcients from fuction of the current distribution
```

Comment:
```suggestion
    # SH coefficients from function of the current distribution
```
comment appears duplicated above